### DISCUSSION TITLE
Sugestion in the documentation

### DISCUSSION THREAD

**Discussion by claudiosa** [vlang/v#14645] [Ideas]
Hi everyone

I hope that this place should be appropriate for a suggestion.
As a newbie in V, I feel missing of "small examples" to illustrate and exhibit how the functions work.
So, the suggestion should be in the documentation more hands-on, where should be found followed for a snippet code.

Surely, the learning curve should be faster.

thanks once

claudio

**Comment by ghost** [comment]
IMHO, one idea is to make use of [examples](https://github.com/vlang/v/tree/master/examples) and make links to it in `docs.md`

**Comment by larpon** [comment]
We have some [examples](https://modules.vlang.io/math.html#digits) - but we need plenty more, that's true

**Comment by claudiosa** [comment]
@Larpon and @kahsa 

My suggestion goes inside of each fn described. For instance, how "rand.string_from_set" function works? I am guessing how it works and could not find an example. 
Really, [https://github.com/vlang/v/tree/master/examples] is useful, but sometimes is hard to find the appropriate example.

I would say, something like that:
.................
x := (  rand.i64_in_range(0,77)  )     // returns a value,  as a function
println('x: ${x}')
.................

Well,  the V is a great language. Keep walking!!!

**Comment by ghost** [comment]
Oh, if you want examples of individual functions, then the function is either badly designed or the function name is too abstract when you don't know how it works without examples.
IMO, it should be rewritten to a function that does minimal work and a function name that you can understand how it works without examples, this to be a matter of the readability of the V language itself.
This is because how can you say that the code is highly readable if you have to check what the function does every time you are reading it?
I understand the desire to have examples when writing, but as far as possible, function and argument names should be easy to understand even without examples.

Nevertheless, some examples are necessary.
My opinion is that the central function of a module, which must have an abstract name, should be described with examples in the module's README.md.

**Comment by larpon** [comment]
> My suggestion goes inside of each fn described. 

I'm not sure I understand this part @claudiosa . I think longer examples still belongs in https://modules.vlang.io/ alongside the function definition or in the header section of the module (the README.md as @kahsa mentioned above) another option is like how [Go has examples](https://pkg.go.dev/log@go1.18#example-Logger) in their docs currently. But before we can have examples somebody must write them :slightly_smiling_face: 

> For instance, how "rand.string_from_set" function works?

Until we have enough examples in the docs an alternative is to use tools like `grep` or `ripgrep` to search for usages in the V project root itself (this is how I personally do it). V's self tests usually has a lot of usage example (although not so "real world" usages :slightly_smiling_face:  )

**Comment by larpon** [comment]
@kahsa 
> ... then the function is either badly designed or the function name is too abstract 

I do not agree with this. Examples is an extremely good way to inspire better coding or give examples of good coding style that will benefit all users - from beginner to pro alike.

Not all functions are simple and can benefit from being explained in detail.

Maybe you come here from another language that has different terms for doing a certain thing e.g.: `printf` vs `println` vs `console.log` vs `log.print` etc. examples makes it easier to understand the equivalent call in V, I think

**Comment by ghost** [comment]
Hmm, I am not talking about not writing a single line of example; it is a matter of culture or principle.
You can call it a convention for readability.

**Comment by larpon** [comment]
> Hmm, I am not talking about not writing a single line of example; it is a matter of culture or principle. You can call it a convention for readability.

@kahsa sorry - I think I might have misunderstood what you wrote 🙂 written communication is not my strong side 😅
